vortex-io/src/object_store.rs

```diff
@@ -124,40 +88,39 @@ impl VortexReadAt for ObjectStoreReadAt {
 }
 
 pub struct ObjectStoreWriter {
-    multipart: Option<WriteMultipart>,
+    upload: Box<dyn MultipartUpload>,
 }
 
 impl ObjectStoreWriter {
-    pub fn new(multipart: WriteMultipart) -> Self {
-        Self {
-            multipart: Some(multipart),
-        }
+    pub async fn new(object_store: Arc<dyn ObjectStore>, location: Path) -> VortexResult<Self> {
+        let upload = object_store.put_multipart(&location).await?;
+        Ok(Self { upload })
     }
 }
 
 impl VortexWrite for ObjectStoreWriter {
     async fn write_all<B: IoBuf>(&mut self, buffer: B) -> io::Result<B> {
-        self.multipart
-            .as_mut()
```

Comment:
this seems to be problematic with `Send`, and anyway nothing here was actually used.